File: Java/AR_354.java
import java.util.ArrayList;
import java.util.Scanner;

public class AR_354 {
    public static int sum(ArrayList<Integer> v) {
        int ret=0;
        
        for (int val: v) {
            ret+=val;
        }
        
        return ret;
    }

    public static boolean solve(Scanner sc) {
        int n=sc.nextInt();
        if (n==0) return false;

        int x=sc.nextInt();
        int maxGlobal=x, maxNinos=x;
        int ninos=1;

        for (int i=1;i<n;i++) {
            x=sc.nextInt();

            if (x>maxGlobal) maxGlobal=x;
            if (x<=maxNinos) {
                ninos=i+1;
                maxNinos=maxGlobal;
            }
        }

        System.out.println(ninos);
        return true;
    }

    // Buffered reader/writer da error RUN_TIME ERROR
    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);

        while (solve(sc));

        sc.close();
    }
}

File: C++/300-400/AR_354.cpp
#include <iostream>
#include <vector>


using namespace std;

// Pre: true
int sum(const vector<int>& v) {
    int r = 0;
    // Inv: 0 <= i <= n and r = Sum k: 0 <= k < i: v[k]
    // T: v.size()-i
    for (int i = 0; i < v.size(); ++i)
        r += v[i];
    return r;
}
// Pos: r = Sum k: 0 <= k < v.size(): v[k]

bool solve() {
    int n;
    cin >> n;	
	if (n == 0)
		return false;

	int x;
	cin >> x;
	int maxGlobal = x, maxNinos = x;
	int ninos = 1;
	int i = 1;
	while (i < n) {
		cin >> x;
		if (x > maxGlobal) maxGlobal = x;
		if (x <= maxNinos) {
			ninos = i + 1;
			maxNinos = maxGlobal;
		}
		i++;
	}


	// write solution
	cout << ninos << '\n';


	return true;
}
int main() {
	while (solve());
	return 0;
}